OpenSim/Actuators/ModelFactory.cpp

```diff
@@ -171,35 +172,40 @@ void ModelFactory::replaceMusclesWithPathActuators(OpenSim::Model &model) {
 
         model.addForce(actu);
 
-        // For the connectee names in the PathPoints to be correct, we must add
-        // the path points after adding the PathActuator to the model.
-        const auto& pathPointSet = musc.getGeometryPath().getPathPointSet();
-        auto& geomPath = actu->updGeometryPath();
-        for (int ip = 0; ip < pathPointSet.getSize(); ++ip) {
-            auto* pathPoint = pathPointSet.get(ip).clone();
-            const auto& socketNames = pathPoint->getSocketNames();
-            for (const auto& socketName : socketNames) {
-                pathPoint->updSocket(socketName)
-                        .connect(pathPointSet.get(ip)
-                                .getSocket(socketName)
-                                .getConnecteeAsObject());
+        // Copy the muscle's path.
```

Comment:
This code's behavior/intent changes with the change.

The function is `ModelFactory::replaceMusclesWithPathActuator`. That should be fine with either point-based or function-based paths. The change here just entirely ignores not-point-based versions of paths, which means that the intent (i.e. to replace all muscles in a model with path actuators) is not upheld.

Reply:
I've added an `else` branch to catch unsupported path types. As we add new path types, we can add new `elseif` branches to included supported path types.